src/main/java/soolutions/hometaskapp/basic/BasicAccount.java

```diff
@@ -0,0 +1,19 @@
+package soolutions.hometaskapp.basic;
+
+import soolutions.hometaskapp.Account;
+import soolutions.hometaskapp.User;
+
+public class BasicAccount implements Account {
+    public BasicAccount(User user) {
```

Comment:
Maybe this is a question for a future ticket. How will the existing Account with a balance be loaded for a user?

Reply:
I think account balance should be implemented with an event stream. This would be more flexible and easier to control. 

Return type on open should be account. Why is boolean desirable? 

basic user and user are differentiated by rules in actuality yes? It should be modeled with rule objects or a state machine.

Reply:
@SilasReinagel to answer your question, I propose a change: `new Accounts(user).iterate()` or some such.

Reply:
@goqp That sounds like a valuable idea. However, it can probably happen on a later ticket, such as when we have Deposits/Withdrawals.

Reply:
Oh okay good point. Its important not to overdesign at each step of the object development.